CommonAI/src/main/java/com/github/sef24sp4/common/ai/IPathfindingProvider.java

```diff
@@ -0,0 +1,20 @@
+package com.github.sef24sp4.common.ai;
+
+import com.github.sef24sp4.common.entities.IEntity;
+import com.github.sef24sp4.common.interfaces.IVector;
+
+public interface IPathfindingProvider {
+
+	/**
+	 * Entity can by implementing this method navigtate to a TargatCoordinate.
+	 * The Method start from the StartCoordinate of the Entity, open neighbor nodes, and expand the node with the lowest fCost.
+	 * When the best route towards a target is known. The Entity will navigate to the next step on this route, which is the TargatCoordinate.
```

Comment:
The text should not have any info regarting the inner workings of the specific AI, if ALL AI's use nodes and fCost, then i am wrong. 
rewrite: Get the next coordinate of a valid path between entity and target coordinate